<launch>

    <!-- Odometry -->
    <arg name="odom_port" default="/dev/ttyACM0"/>
    <arg name="use_odom" default="true"/>

    <group if="$(arg use_odom)">
        <node pkg="horiokart2021_sensors" type="serial_odom" name="serial_odom" output="screen">
            <param name="device_name" type="string" value="$(arg odom_port)" />
            <param name="odom_frame" type="string" value="odom" />
            <param name="base_frame" type="string" value="base_footprint" />
            <param name="odom_pub_rate" type="int" value="20" />

            <param name="inv_x" type="bool" value="true" />
            <param name="inv_y" type="bool" value="true" />
            <param name="inv_th" type="bool" value="false" />

            <param name="csv" type="bool" value="true" />
        </node>

        <node pkg="horiokart2021_sensors" type="hj_sensor_odom2tf" name="odom2tf">
        </node>
    </group>

    <!-- rplidar -->
    <arg name="front_rplidar_port" default="/dev/ttyUSB0"/>
    <arg name="top_rplidar_port" default="/dev/ttyUSB1"/>
    <arg name="use_rplidar" default="true"/>

    <group if="$(arg use_rplidar)">
        <!-- rplidar A1 -->
        <node name="front_rplidarNode" pkg="rplidar_ros"  type="rplidarNode" output="screen">
            <param name="serial_port"         type="string" value="$(arg front_rplidar_port)"/>
            <param name="serial_baudrate"     type="int"    value="115200"/><!--A1/A2 -->
            <!--param name="serial_baudrate"     type="int"    value="256000"--><!--A3 -->
            <param name="frame_id"            type="string" value="front_lrf_link"/>
            <param name="inverted"            type="bool"   value="false"/>
            <param name="angle_compensate"    type="bool"   value="true"/>
        </node>

        <!-- rplidar S2 -->
        <node name="top_rplidarNode" pkg="rplidar_ros"  type="rplidarNode" output="screen">
            <param name="serial_port"         type="string" value="$(arg top_rplidar_port)"/>
            <param name="serial_baudrate"     type="int"    value="1000000"/>
            <param name="frame_id"            type="string" value="top_lrf_link"/>
            <param name="inverted"            type="bool"   value="false"/>
            <param name="angle_compensate"    type="bool"   value="true"/>
        </node>

        <include file="$(find horiokart2021_sensors)/launch/laser_filter.launch"></include>

    </group>

    
</launch>
